src/screens/map-screen/map-screen.component.tsx

```diff
@@ -189,11 +201,67 @@ const MapScreen = () => {
    */
   useEffect(() => {
     setCurrentRegion(getCampusById(CampusId.SGW).region);
+    if (!currentLocation) {
+      getCurrentLocationAsync().then(location => {
```

Comment:
I don't think we need the user's location at all times. Consider only calling this method when you actually need the location (like when starting a travel plan

Reply:
Also, please add a catch() for every call to getCurrentLocationAsync